### DISCUSSION TITLE
尝鲜试了下，似乎速度并没有什么提升

### DISCUSSION THREAD

**Discussion by anerg2046** [XTLS/Xray-core#61] [Performance]
xray+xtls+tcp的配置，同一机房，相同的两台vps
一台是docker跑的caddy+v2ray+ws，另一台是ProxySU安装的vless+xtls(direct)+tcp
看youtube的时候两者速度相差无几，甚至ws的还要快一点

个人感觉xtls的优势在于软路由cpu占用略低5%，但速度上并无太大差别

**Comment by RPRX** [comment]
XTLS 无法突破线路瓶颈，建议了解这些协议的底层原理，测速建议 Speedtest

**Comment by anerg2046** [comment]
明白你的意思，目前对于大家都能跑满的情况下，其实用什么区别不大，特别是在家用软路由或者台式机软件的情况下。
xtls的优势可能在于移动设备在外出情况下的省电

**Comment by RPRX** [comment]
并不是都能跑满，之前很多人从硬路由到软路由，主要是因为前者性能不够，XTLS 可以改变这一状况

其中一些人是喜欢废物利用 DIY，但专门买软路由似乎没有什么必要了

**Comment by ghost** [comment]
用两台VPS分别部署VLESS+XTLS和VMess+tls交叉测试，你就会有结果了

**Comment by anerg2046** [comment]
我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派

**Comment by ghost** [comment]
XTLS在树莓派上对速度的提升是有目共睹的

尤其是在树莓派上运行服务端

**Comment by anerg2046** [comment]
> 用两台VPS分别部署VLESS+XTLS和VMess+tls交叉测试，你就会有结果了

相同机房，相同配置，还要交叉测试，我觉得就是抬杠了。

**Comment by anerg2046** [comment]
> XTLS在树莓派上对速度的提升是有目共睹的
> 
> 尤其是在树莓派上运行服务端

是的，对cpu占用要低。不过一般都是跑软路由，运行服务端干嘛

**Comment by anerg2046** [comment]
对了，还有个比较麻烦的点，就是需要接管443，这样就没法caddy自动申请证书了。
因为我习惯把东西都放docker里，让宿主机相对干净。
目前好像caddy还无法支持类似nginx的ssl_preread

**Comment by ghost** [comment]
> > 用两台VPS分别部署VLESS+XTLS和VMess+tls交叉测试，你就会有结果了
> 
> 相同机房，相同配置，还要交叉测试，我觉得就是抬杠了。

首先，测试时要排除其他因素干扰，从你的描述来看，你并没有对线路进行控制

或者说，被测试的两个对象都已经达到你的线路瓶颈

我仅仅是给你提供了一个相对更加合理的测试方案，并不是抬杠

**Comment by niuniudada** [comment]
介绍里也是说性能上有提升，并没提到过会提升速度。速度快慢关键还是看线路

**Comment by simpleandstupid** [comment]
youtube测速......
这是认真的吗，影响因素太多了（比如这个视频它只需要这个吞吐量）
xtls是节省二次加密的开销，如果性能绰绰有余速度不会有太大提升。毕竟都是要讲科学的（又不是魔法），都能跑到本地带宽上限那比较也没啥意义了。

**Comment by shenlijun** [comment]
youtube测试看不出来的，就算是这样，youtube也到不了20W，当然我的机器带不动8k加上2倍速。8k+1.25倍也就12W左右
![无标题](https://user-images.githubusercontent.com/8860591/101886610-b4f61b80-3bd6-11eb-88a0-611817c6a6ef.png)

**Comment by shenlijun** [comment]
> 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派

不知4b可否跑慢千兆？手头马上会有一个4b，近期搬家考虑新配网关，要是4b可以就不买新的了

**Comment by anerg2046** [comment]
> 我现在可以跑960m
> 
> > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> 
> 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关

个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
我家里是600M联通光纤，测速能到570M的样子

PS：youtube看4K的话，2W就够了

**Comment by shenlijun** [comment]
> > 我现在可以跑960m
> > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > 
> > 
> > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> 
> 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> 我家里是600M联通光纤，测速能到570M的样子
> 
> PS：youtube看4K的话，2W就够了

我是个跑跑党-_-#

**Comment by anerg2046** [comment]
> youtube测试看不出来的，就算是这样，youtube也到不了20W，当然我的机器带不动8k加上2倍速。8k+1.25倍也就12W左右
> ![无标题](https://user-images.githubusercontent.com/8860591/101886610-b4f61b80-3bd6-11eb-88a0-611817c6a6ef.png)

youtube的测速其实才是更为准确的，因为这才是你真正的使用场景。speedtest.net的原理是根据ip找到和你ip的地理位置最近的一个测试服务器进行连接测试

**Comment by anerg2046** [comment]
> > > 我现在可以跑960m
> > > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > > 
> > > 
> > > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> > 
> > 
> > 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> > 我家里是600M联通光纤，测速能到570M的样子
> > PS：youtube看4K的话，2W就够了
> 
> 我是个跑跑党-_-#

以前我也是，后来发现毫无必要，现在主力vps都是便宜的，特别是俄罗斯那个，不限流量，一年才100出头，看4k足够足够。就网络只有200M

**Comment by RPRX** [comment]
@anerg2046 

这里指的是当你要比较协议的客观表现时，Speedtest 是一个更好的选择，因为 Youtube 会受到加载及缓存策略的影响，浮动较大

**Comment by 2444989513** [comment]
我只能说用youtube测速的都是辣鸡

**Comment by j4Uq** [comment]
> > > > 我现在可以跑960m
> > > > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > > > 
> > > > 
> > > > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> > > 
> > > 
> > > 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> > > 我家里是600M联通光纤，测速能到570M的样子
> > > PS：youtube看4K的话，2W就够了
> > 
> > 
> > 我是个跑跑党-_-#
> 
> 以前我也是，后来发现毫无必要，现在主力vps都是便宜的，特别是俄罗斯那个，不限流量，一年才100出头，看4k足够足够。就网络只有200M

能透露下是哪家俄罗斯的吗，我的刚好快到期了，正犹豫买哪家的

**Comment by shenlijun** [comment]
> > > > 我现在可以跑960m
> > > > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > > > 
> > > > 
> > > > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> > > 
> > > 
> > > 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> > > 我家里是600M联通光纤，测速能到570M的样子
> > > PS：youtube看4K的话，2W就够了
> > 
> > 
> > 我是个跑跑党-_-#
> 
> 以前我也是，后来发现毫无必要，现在主力vps都是便宜的，特别是俄罗斯那个，不限流量，一年才100出头，看4k足够足够。就网络只有200M

便宜、不限流量，带宽标的还可以
这就跟小区宽带一样，说起来溜得飞起，用起来一言难尽

> > > > > 我现在可以跑960m
> > > > > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > > > > 
> > > > > 
> > > > > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> > > > 
> > > > 
> > > > 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> > > > 我家里是600M联通光纤，测速能到570M的样子
> > > > PS：youtube看4K的话，2W就够了
> > > 
> > > 
> > > 我是个跑跑党-_-#
> > 
> > 
> > 以前我也是，后来发现毫无必要，现在主力vps都是便宜的，特别是俄罗斯那个，不限流量，一年才100出头，看4k足够足够。就网络只有200M
> 
> 能透露下是哪家俄罗斯的吗，我的刚好快到期了，正犹豫买哪家的

便宜、不限流量，带宽标的还可以
这就跟小区宽带一样，说起来溜得飞起，用起来一言难尽

**Comment by shenlijun** [comment]
> > youtube测试看不出来的，就算是这样，youtube也到不了20W，当然我的机器带不动8k加上2倍速。8k+1.25倍也就12W左右
> > ![无标题](https://user-images.githubusercontent.com/8860591/101886610-b4f61b80-3bd6-11eb-88a0-611817c6a6ef.png)
> 
> youtube的测速其实才是更为准确的，因为这才是你真正的使用场景。speedtest.net的原理是根据ip找到和你ip的地理位置最近的一个测试服务器进行连接测试

它找的是vps附近的进行测试
但是测试结果却能从侧面表征你本地到vps的延迟和带宽，就是线路质量的好坏
一般国外的vps只要不垃圾，它到youtube都没什么问题
大部分时候影响体验的是自己的线路

**Comment by anerg2046** [comment]
> > > > > 我现在可以跑960m
> > > > > > 我用的树莓派做单臂，600M联通可以跑满，不过似乎很多人不知道怎么玩树莓派
> > > > > 
> > > > > 
> > > > > 不知4b可否跑慢千兆？手头上有一个4b，近期搬家考虑新配网关
> > > > 
> > > > 
> > > > 个人认为需要跑满的需求并不大，能流畅看4k就足够了。树莓派做单臂绰绰有余，我是3B+，你的4B应该更没问题了。
> > > > 我家里是600M联通光纤，测速能到570M的样子
> > > > PS：youtube看4K的话，2W就够了
> > > 
> > > 
> > > 我是个跑跑党-_-#
> > 
> > 
> > 以前我也是，后来发现毫无必要，现在主力vps都是便宜的，特别是俄罗斯那个，不限流量，一年才100出头，看4k足够足够。就网络只有200M
> 
> 能透露下是哪家俄罗斯的吗，我的刚好快到期了，正犹豫买哪家的

北方联通可以试试，其他地方我不清楚，带宽200，一般speedtest跑出来是180的样子
justhost.ru
[带我的推荐码链接](https://justhost.ru/?ref=69692)

@shenlijun 
其实便宜的也不一定就不好，还是看自身网络情况，北方联通走俄罗斯来回都是cn2，油管平均2.5w，足够我看4k啦，关键是真的便宜。。而且不限流量

**Reply by shenlijun** [reply]
这个要看高峰期的延迟带宽和丢包率

**Reply by hyyz17200** [reply]
> 这个要看高峰期的延迟带宽和丢包率

是的，不过这家对联通较友好，一直在用。

**Comment by anerg2046** [comment]
@rprx 是的，对协议来说，应该采用speedtest，不过考虑到实际场景，就选择了看油管的速度

@2444989513 兄弟，我不想跟你杠，但是你在科学上网的时候是只去跑speedtest还是说去看视频？

**Comment by zhuzeitou** [comment]
> @2444989513 兄弟，我不想跟你杠，但是你在科学上网的时候是只去跑speedtest还是说去看视频？

典型的“我用不到所以没用”，拿一个受一堆其他因素制约的方案得到一个没意义的所谓“差不多”的结论，不如看看人家是如何[测试](https://github.com/badO1a5A90/v2ray-doc)的

**Comment by SekiBetu** [comment]


**Reply by ghost** [reply]
哈哈，和谐交流

（大雾

**Reply by ghost** [reply]
他的原话：“相同机房，相同配置，还要交叉测试，我觉得就是抬杠了。”

我也是很无语

**Reply by shenlijun** [reply]
还有“不过似乎很多人不知道怎么玩树莓派”，呃……

**Comment by RPRX** [comment]
（友好客观交流
